cmd/thv-operator/controllers/mcpregistry_controller.go

```diff
@@ -211,16 +392,443 @@ func (r *MCPRegistryReconciler) finalizeMCPRegistry(ctx context.Context, registr
 	}
 
 	// TODO: Add additional cleanup logic when other features are implemented:
-	// - Clean up Registry API service
+	// - Clean up Registry API deployment and service (will be handled by owner references)
 	// - Cancel any running sync operations
 
 	ctxLogger.Info("MCPRegistry finalization completed", "registry", registry.Name)
 	return nil
 }
 
+// labelsForRegistryAPI generates standard labels for registry API resources
+func labelsForRegistryAPI(mcpRegistry *mcpv1alpha1.MCPRegistry, resourceName string) map[string]string {
+	return map[string]string{
+		"app.kubernetes.io/name":             resourceName,
+		"app.kubernetes.io/component":        "registry-api",
+		"app.kubernetes.io/managed-by":       "toolhive-operator",
+		"toolhive.stacklok.io/registry-name": mcpRegistry.Name,
+	}
+}
+
+// buildRegistryAPIDeployment creates and configures a Deployment object for the registry API.
+// This function handles all deployment configuration including labels, container specs, probes,
+// and storage manager integration. It returns a fully configured deployment ready for Kubernetes API operations.
+func (r *MCPRegistryReconciler) buildRegistryAPIDeployment(
+	mcpRegistry *mcpv1alpha1.MCPRegistry, sourceHandler sources.SourceHandler,
+) (*appsv1.Deployment, error) {
+	// Generate deployment name using the established pattern
+	deploymentName := fmt.Sprintf("%s-api", mcpRegistry.Name)
+
+	// Define labels using common function
+	labels := labelsForRegistryAPI(mcpRegistry, deploymentName)
+
+	// Create basic deployment specification with named container
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deploymentName,
+			Namespace: mcpRegistry.Namespace,
+			Labels:    labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &[]int32{1}[0], // Single replica for registry API
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app.kubernetes.io/name":      deploymentName,
+					"app.kubernetes.io/component": "registry-api",
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+					Annotations: map[string]string{
+						"toolhive.stacklok.dev/config-hash": r.getSourceDataHash(mcpRegistry, sourceHandler),
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: "toolhive-registry-api",
+					Containers: []corev1.Container{
+						{
+							Name:  registryAPIContainerName,
+							Image: getRegistryAPIImage(),
+							Args: []string{
+								"serve",
+							},
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 8080,
```

Comment:
use constants for 8080 and similar embedded values? (e.g. resource limits and requests)

Reply:
done